scene/animation/animation_mixer.cpp

```diff
@@ -954,15 +954,15 @@ Variant AnimationMixer::post_process_key_value(const Ref<Animation> &p_anim, int
 
 Variant AnimationMixer::_post_process_key_value(const Ref<Animation> &p_anim, int p_track, Variant p_value, const Object *p_object, int p_object_idx) {
 	switch (p_anim->track_get_type(p_track)) {
-#ifndef _3D_DISABLED
 		case Animation::TYPE_POSITION_3D: {
+#ifndef _3D_DISABLED
 			if (p_object_idx >= 0) {
 				const Skeleton3D *skel = Object::cast_to<Skeleton3D>(p_object);
 				return Vector3(p_value) * skel->get_motion_scale();
 			}
 			return p_value;
-		} break;
 #endif // _3D_DISABLED
+		} break;
 		default: {
 		} break;
 	}
```

Comment:
Considering we are using only one of the track type values, this function might make more sense if we just rewrite it to not use a switch statement.

```cpp
Variant AnimationMixer::_post_process_key_value(const Ref<Animation> &p_anim, int p_track, Variant p_value, const Object *p_object, int p_object_idx) {
#ifndef _3D_DISABLED
	if (p_anim->track_get_type(p_track) == Animation::TYPE_POSITION_3D && p_object_idx >= 0) {
		const Skeleton3D *skel = Object::cast_to<Skeleton3D>(p_object);
		return Vector3(p_value) * skel->get_motion_scale();
	}
#endif // _3D_DISABLED
	return p_value;
}
```